src/error.ts

```diff
@@ -99,25 +87,37 @@ export function operationAborted (context: string, reason: string) {
   return errCode(new OperationAbortedError(context, reason), codes.ERR_ALREADY_ABORTED)
 }
 
-export class DataChannelError extends WebRTCTransportError {
-  constructor (streamLabel: string, errorMessage: string) {
-    super(`[stream: ${streamLabel}] data channel error: ${errorMessage}`)
-    this.name = 'WebRTC/DataChannelError'
+export class OverStreamLimitError extends WebRTCTransportError {
+  constructor (msg: string) {
+    super(msg)
+    this.name = 'WebRTC/OverStreamLimitError'
   }
 }
 
-export function dataChannelError (streamLabel: string, msg: string) {
-  return errCode(new DataChannelError(streamLabel, msg), codes.ERR_DATA_CHANNEL)
+export function overStreamLimit (dir: Direction, proto: string) {
+  const code = dir === 'inbound' ? codes.ERR_TOO_MANY_INBOUND_PROTOCOL_STREAMS : codes.ERR_TOO_MANY_OUTBOUND_PROTOCOL_STREAMS
+  return errCode(new OverStreamLimitError(`${dir} stream limit reached for protocol - ${proto}`), code)
 }
 
-export class StreamingLimitationError extends WebRTCTransportError {
-  constructor (msg: string) {
+export class UnimplementedError extends WebRTCTransportError {
+  constructor (methodName: string) {
+    super('A method (' + methodName + ') was called though it has been intentionally left unimplemented.')
```

Comment:
isn't it frowned upon to not use template strings, now that these are widely available?

Reply:
Good call, this pattern is updated throughout this file.